cli/waiter/subcommands/maintenance.py

```diff
@@ -0,0 +1,81 @@
+import requests
+
+from waiter import terminal, http_util
+from waiter.token_post import process_post_result, post_failed_message
+from waiter.querying import get_cluster_with_token, get_token
+from waiter.util import guard_no_cluster, logging, print_info
+
+
+def maintenance(clusters, args, _):
+    guard_no_cluster(clusters)
+    logging.debug('args: %s' % args)
+    token_name = args['token']
+    sub_action = args['sub_action']
+    cluster = get_cluster_with_token(clusters, token_name)
+    cluster_name = cluster['name']
+    cluster_url = cluster['url']
+    existing_token_data, existing_token_etag = get_token(cluster, token_name)
+    maintenance_mode_active = 'maintenance' in existing_token_data
+    json_body = existing_token_data
+    headers = {'If-Match': existing_token_etag}
+    params = {'token': token_name}
+    try:
+        if sub_action == 'check':
+            print_info(f'{token_name} is {"" if maintenance_mode_active else "not "}in maintenance mode')
+            return 0 if maintenance_mode_active else 1
+        elif sub_action == 'start':
+            update_doc = {"maintenance": {"message": args['message']}}
+            json_body.update(update_doc)
+        elif sub_action == 'stop':
+            if maintenance_mode_active:
+                json_body.pop("maintenance")
+            else:
+                raise Exception("Token is not in maintenance mode")
+        resp = http_util.post(cluster, 'token', json_body, params=params, headers=headers)
+        process_post_result(resp)
+        return 0
+    except requests.exceptions.ReadTimeout as rt:
+        logging.exception(rt)
+        print_info(terminal.failed(
+            f'Encountered read timeout with {cluster_name} ({cluster_url}). The operation may have completed.'))
+        return 1
+    except IOError as ioe:
+        logging.exception(ioe)
+        reason = f'Cannot connect to {cluster_name} ({cluster_url})'
+        message = post_failed_message(cluster_name, reason)
+        print_info(f'{message}\n')
+
+
+def register_action(add_parser, sub_action):
+    parser = add_parser(sub_action, help=f'{sub_action} maintenance mode for a token')
+    parser.add_argument('token')
+    parser.set_defaults(sub_action=sub_action)
+    return parser
+
+
+def register_check(add_parser):
+    register_action(add_parser, 'check')
+
+
+def register_stop(add_parser):
+    register_action(add_parser, 'stop')
+
+
+def register_start(add_parser):
+    start_parser = register_action(add_parser, 'start')
+    start_parser.add_argument('--message', '-m',
```

Comment:
Please make this a positional argument since it is required.

Reply:
good catch

Reply:
![Screenshot from 2020-11-12 19-21-40](https://user-images.githubusercontent.com/8290559/99016268-5a589980-251c-11eb-97f6-e1d5160be190.png)

This is the output whenever users put "waiter maintenance" without a sub action (required positional argument)

Reply:
Why does it say `cmd` is required?

Reply:
By default argparse gives the message, "waiter maintenance: error: the following arguments are required: [dest]". We can't change this behavior unless we override the internal implementation for a custom error message ([source](https://stackoverflow.com/questions/30956688/python-argparse-customizing-error-messages)).

1. Does "subcommand" seem better?
![Screenshot from 2020-11-12 22-08-14](https://user-images.githubusercontent.com/8290559/99027872-ecb86780-2533-11eb-9350-b840d280d2a8.png)
2. Or should we just make it display the help message like the `waiter ` command with no sub command:
![Screenshot from 2020-11-12 22-12-54](https://user-images.githubusercontent.com/8290559/99027991-4325a600-2534-11eb-94ef-f8c7c35478a0.png)
![Screenshot from 2020-11-12 22-12-37](https://user-images.githubusercontent.com/8290559/99027994-4456d300-2534-11eb-8f85-c389a8f60d2b.png)

Reply:
Talked offline, and we are going with the help message for consistency